### PR TITLE
Fix index pages for Runtime and Provider API docs

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/339.

It wasn't intentional that we didn't have `index.md` files and instead used `/runtime_service` and `/ibm_provider`. It was only because that's what the original qiskit.org docs do, and I don't think we realized that was bad and we should fix it.

This PR does mean `/runtime_service` and `/ibm_provider` will now 404. I couldn't find any references to those pages outside of our own docs, so it didn't seem necessary to set up redirects. But I can if we think it's important.

### CODE REVIEW

File: docs/api/qiskit-ibm-provider/_toc.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
       "children": [
         {
           "title": "Overview",
-          "url": "/api/qiskit-ibm-provider/ibm_provider"
+          "url": "/api/qiskit-ibm-provider/index"
```
Comment (frankharkins) [inline docs/api/qiskit-ibm-provider/_toc.json:9]: Should we maybe link to just `/api/qiskit-ibm-provider/` (i.e. without `/index`)?

This is a bikeshed comment though so just go with whatever you prefer.

**Review by frankharkins — COMMENTED** [review]
Looks good! Are you also updating the links in the nav bar?

Screenshot 2023-11-15 at 15 04 40

File: docs/api/qiskit-ibm-provider/_toc.json (hunk 1 above)
Comment (Eric-Arellano) [reply docs/api/qiskit-ibm-provider/_toc.json:9]: Eh, couldn't figure out quickly how to do this so I'm going to punt for now: https://github.com/Qiskit/documentation/issues/361. It's behaving this way for Qiskit too

**Review by Eric-Arellano — COMMENTED** [review]
Yes to updating the top nav bar and home page when syncing these docs to go live. Good catch!

**Review by frankharkins — APPROVED** [review]
Nice, thanks!

**Pull request merged**